### PR TITLE
add venv to docker and refactor

### PR DESCRIPTION
**Pull request by psych0d0g**
introduce python virtual environments so that project dependencies dont clash with the OS shipped python libraries.
As an example poetry tried to update `packaging` which is not possible since its usually shipped with the used OS distro and can not be managed by pip or poetry if installed in the OS, with venvs that limitation is solved and said package is able to be managed by the project.
with a venv we are indenpendent from what the OS shipps, even in core libraries and can update them however we want :)

i also added some meaningful comments to the dockerfile to explain what is happening

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -38,12 +34,18 @@ passlib = {extras = ["bcrypt"], version = "^1.7.4"}
...
 python-multipart = "^0.0.6"
 types-python-jose = "^3.3.4.8"
 types-passlib = "^1.7.7.13"
 
+[tool.poetry.group.test.dependencies]
+pytest = "^7.4.0"
+pytest-env = "^0.8.2"
+pytest-vcr = "^1.0.2"
+pytest-mock = "^3.11.1"
+pytest-asyncio = "^0.21.1"
```
Comment (gantoine) [inline pyproject.toml:47]: 👍🏼

File: docker/Dockerfile [hunk 2]
```diff
@@ -29,31 +28,69 @@ RUN apk update && \
...
+    /installed_poetry_requirements.txt \
+    /build_requirements.txt \
+    /docker-entrypoint.sh \
+    /docker-entrypoint.d
+
+# Move everything we prepared over to our final docker image
 FROM scratch
+LABEL org.opencontainers.image.title="zurdi15/romm" \
+      org.opencontainers.image.description="RomM (stands for Rom Manager) is a game library manager focused in retro gaming. Manage and organize all of your games from a web browser" \
+      org.opencontainers.image.licenses="GPL-3.0"
```
Comment (gantoine) [inline docker/Dockerfile]: 👍🏼

**Review by GAntoine — APPROVED** [review]
yeah really solid 👍🏼

File: docker/Dockerfile (hunk 2 above)
Comment (Djiit) [reply docker/Dockerfile]: FWIW, the docker metadata action in the `build` workflow should be taking care of title and description labels automatically. Maybe add the `description` label [here](https://github.com/zurdi15/romm/blob/release/.github/workflows/build.yml#L40C10-L40C10) too ?

Comment (psych0d0g) [reply docker/Dockerfile]: moved the labels into the git workflow file as suggested

**Review by zurdi15 — APPROVED** [review]

**Pull request merged**